### PR TITLE
resolve storybook config to Framework

### PR DESCRIPTION
**Pull request by thedavidprice**
@peterp I based this change off your update to `dev` here:
https://github.com/redwoodjs/redwood/commit/45bf2c919de1900926f2040e25e29f7faa7089e3#diff-65e03eec28ca193290677c0c09dc9f9f9e180d41613f6bc324e8f94e8ed95094

Haven't tested it out yet. 😬

### CODE REVIEW

File: packages/cli/src/commands/storybook.js [hunk 1]
```diff
@@ -44,11 +44,15 @@ export const handler = ({ open, port, build, buildDirectory }) => {
     shell: true,
     cwd,
   })
+
```
Comment (thedavidprice) [inline packages/cli/src/commands/storybook.js:47]:
```suggestion
```

File: packages/cli/src/commands/storybook.js [hunk 2]
```diff
@@ -44,11 +44,15 @@ export const handler = ({ open, port, build, buildDirectory }) => {
...
+  
+  const storybookConfig = require.resolve(
+     '@redwoodjs/testing/config/storybook'
+   )  
 
   execa(
     `yarn ${build ? 'build' : 'start'}-storybook`,
     [
-      '--config-dir ../node_modules/@redwoodjs/testing/config/storybook',
+      '--config-dir "${storybookConfig}"',
```
Comment (thedavidprice) [inline packages/cli/src/commands/storybook.js:55]:
```suggestion
      `--config-dir "${storybookConfig}"`,
```

File: packages/cli/src/commands/storybook.js [hunk 3]
```diff
@@ -44,11 +44,15 @@ export const handler = ({ open, port, build, buildDirectory }) => {
     shell: true,
     cwd,
   })
+  
+  const storybookConfig = require.resolve(
+     '@redwoodjs/testing/config/storybook'
+   )
```
Comment (thedavidprice) [inline packages/cli/src/commands/storybook.js:50]:
```suggestion
  const storybookConfig = require.resolve('@redwoodjs/testing/config/storybook')  
```

**Comment by thedavidprice** [comment]
@peterp I just coded this change directly here on GitHub. Seeing now that Storybook isn't passing E2E. I'll take a look at it next Monday unless you want to resolve and merge first — go for it if so.

**Comment by peterp** [comment]
@thedavidprice I'm taking a look now.

**Comment by peterp** [comment]
@thedavidprice This should work now.

**Comment by peterp** [comment]
oh, let me me make it use the relative binaries as well.

**Comment by thedavidprice** [comment]
@peterp huge thanks for moving this forward! I can confirm it starts and builds successfully. However, the stories themselves won't load in either case. I'm assuming it's related to something with the Storybook Provider.

Possible you also need to handle `~__REDWOOD__USER_ROUTES_FOR_MOCK` in MockProviders?
https://github.com/redwoodjs/redwood/blob/31a9ab01a7ed426ccc2d876d9fb87a801200acfc/packages/testing/src/MockProviders.tsx#L19

Screenshots of error for both storybook run and build:
Screen Shot 2021-06-28 at 5 09 40 PM

Screen Shot 2021-06-28 at 5 09 48 PM

**Comment by thedavidprice** [comment]
@peterp I've upgraded to Storybook v6.3.1 with no problems. There's a lot of focus in SB 6.2 and 6.3 on package managers and bundlers/builders. Check out the release announcements:
- https://storybook.js.org/blog/storybook-6-3/
- https://storybook.js.org/blog/storybook-6-2/

Changelog/Migration Guide:
- [From version 6.1.x to 6.2.0](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-61x-to-620)
- [From version 6.2.x to 6.3.0](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-62x-to-630)

## Breaking
There are no breaking changes in Redwood packages. However, both 6.2 and 6.3 contain breaking changes and deprecations that may affect some user implementations/configuration.

**Comment by peterp** [comment]
@thedavidprice I think this may be an issue:

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated-storyfn

I think _we_ use this storyFn somewhere.

**Comment by peterp** [comment]
But happy for this to merge in the meantime.

**Comment by thedavidprice** [comment]
@peterp I might not have been clear that there is still an issue using --cwd after upgrading to 6.3

Were you able to diagnose this? https://github.com/redwoodjs/redwood/pull/2892#issuecomment-870129466

Also, understood and agreed about our use case for SB. But I do think there's a near-term advantage to us using established tools. (Trying to help us feel like all this upgrading is worth it!) Let's keep noodling post v1...

**Comment by peterp** [comment]
> I might not have been clear that there is still an issue using --cwd after upgrading to 6.3
@thedavidprice got it 👍 
Let's upgrade and then I'll try and fix `--cwd` - it's referencing the wrong binary!

**Comment by thedavidprice** [comment]
@peterp I wasn't entirely sure what you meant, so I went ahead and upgraded Storybook via #2940 Just merged in changes from main branch.

FWIW, the only way I was able to get this working for testing was to use `yarn rwt link`. Add the SB dependencies to the web/ workspace in my test project didn't work — it was still using SB v6.1. I believe this is what you were referring to.

Now that we can upgrade a test project to canary with SB v6.3, will that help here?

**Review by peterp — APPROVED** [review]
@thedavidprice I think this is good to go.

**Comment by thedavidprice** [comment]
Merged. I'll test on canary and see if the Story issue is resolved.

**Pull request merged**